ogre2/src/Ogre2ThermalCamera.cc

```diff
@@ -231,6 +257,102 @@ void Ogre2ThermalCameraMaterialSwitcher::preRenderTargetUpdate(
+          }
+          else
+          {
+            baseName = common::basename(texture);
+            auto idx = texture.rfind(baseName);
+            if (idx != std::string::npos)
+            {
+              std::string dirPath = texture.substr(0, idx);
+              if (!dirPath.empty())
+              {
+                foundTexture = true;
+                if (!Ogre::ResourceGroupManager::getSingleton().
+                      resourceLocationExists(dirPath))
+                {
+                  Ogre::ResourceGroupManager::getSingleton().addResourceLocation(
+                      dirPath, "FileSystem", "General");
+                }
+              }
+            }
+          }
+
+          // create a material for this item, now that the texture has been
+          // searched for. We must clone the base heat signature material since
+          // different items may use different textures. We also append the
+          // item's ID to the end of the new material name to ensure new
+          // material uniqueness in case two items use the same heat signature
+          // texture, but have different temperature ranges
+          auto heatSignatureMaterial = this->baseHeatSigMaterial->clone(
+              this->name + "_" + baseName + "_" +
+              Ogre::StringConverter::toString(item->getId()));
+          if (foundTexture)
+          {
+            auto textureUnitStatePtr = heatSignatureMaterial->
+              getTechnique(0)->getPass(0)->getTextureUnitState(0);
+            Ogre::String textureName = baseName;
+            textureUnitStatePtr->setTextureName(textureName);
+          }
+          else
+          {
+            ignerr << "Error loading texture: " << texture << "\n"
+              << "(material will be loaded without a texture applied to it)\n";
+          }
+          // set temperature range for the heat signature
+          auto minTempVariant = ogreVisual->UserData("minTemp");
+          auto maxTempVariant = ogreVisual->UserData("maxTemp");
+          auto minTemperature = std::get_if<float>(&minTempVariant);
+          auto maxTemperature = std::get_if<float>(&maxTempVariant);
+          if (minTemperature && maxTemperature)
+          {
+            // TODO(adlarkin) check for bad input
+            // (negative temps, values above the camera's max kelvin, etc)
+            Ogre::GpuProgramParametersSharedPtr params =
+              heatSignatureMaterial->getTechnique(0)->getPass(0)->
+              getFragmentProgramParameters();
+            params->setNamedConstant("minTemp",
+                static_cast<float>(*minTemperature));
+            params->setNamedConstant("maxTemp",
+                static_cast<float>(*maxTemperature));
+          }
+          heatSignatureMaterial->load();
+          this->heatSignatureMaterials[item->getId()] = heatSignatureMaterial;
+        }
+
+        // set visibility flag so thermal camera can see it
+        item->addVisibilityFlags(0x10000000);
```

Comment:
Should this flag, `0x10000000`, be a static variable or part of an enum for clarity?

Reply:
This flag should have been set to [`IGN_VISIBILITY_SELECTION`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering4/include/ignition/rendering/RenderTypes.hh#L28-L30). However, now that this PR uses #211, visibility flags aren't used anymore, so this is no longer an issue.